src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapGet_NoParam_StringReturn_WithFilter.generated.txt

```diff
@@ -0,0 +1,206 @@
+
+    %GENERATEDCODEATTRIBUTE%
+internal static global::Microsoft.AspNetCore.Builder.RouteHandlerBuilder MapGet(
+        this global::Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints,
+        [global::System.Diagnostics.CodeAnalysis.StringSyntax("Route")] string pattern,
+        global::System.Func<string> handler,
+        [global::System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [global::System.Runtime.CompilerServices.CallerLineNumber]int lineNumber = 0)
+    {
+        return global::Microsoft.AspNetCore.Http.Generated.GeneratedRouteBuilderExtensionsCore.MapCore(endpoints, pattern, handler, GetVerb, filePath, lineNumber);
+    }
+
+}
+
+namespace Microsoft.AspNetCore.Http.Generated
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using System.IO;
+    using Microsoft.AspNetCore.Routing;
+    using Microsoft.AspNetCore.Routing.Patterns;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.FileProviders;
+    using Microsoft.Extensions.Primitives;
+
+    using MetadataPopulator = System.Func<System.Reflection.MethodInfo, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions?, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult>;
+    using RequestDelegateFactoryFunc = System.Func<System.Delegate, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult?, Microsoft.AspNetCore.Http.RequestDelegateResult>;
+
+    %GENERATEDCODEATTRIBUTE%
+    file static class GeneratedRouteBuilderExtensionsCore
+    {
+        internal static class GenericThunks<T>
+        {
+            public static readonly System.Collections.Generic.Dictionary<(string, int), (MetadataPopulator, RequestDelegateFactoryFunc)> map = new()
+            {
+
+            };
+        }
+
+        internal static readonly System.Collections.Generic.Dictionary<(string, int), (MetadataPopulator, RequestDelegateFactoryFunc)> map = new()
+        {
+                        [(@"TestMapActions.cs", 15)] = (
+           (methodInfo, options) =>
+            {
+                if (options == null)
+                {
+                    return new RequestDelegateMetadataResult { EndpointMetadata = ReadOnlyCollection<object>.Empty };
+                }
+                options.EndpointBuilder.Metadata.Add(new SourceKey(@"TestMapActions.cs", 15));
+                return new RequestDelegateMetadataResult { EndpointMetadata = options.EndpointBuilder.Metadata.AsReadOnly() };
+            },
+            (del, options, inferredMetadataResult) =>
+            {
+                var handler = (System.Func<string>)del;
```

Comment:
```suggestion
                var handler = (Func<string>)del;
```

Reply:
This is the only fully-qualified reference I wasn't able to remove. There's one function that generates the delegate definition that's used in both the body of the RequestDelegateGenerator and in the strongly-typed `MapAction` overload. Since the `MapAction` overload needs the fully qualified type we do that in both places. It's not a very nested type so hopefully this is fine.

Reply:
I think this is OK, I'm not too concerned about it.

If we did want to eliminate this, `StaticRouteHandlerModelEmitter.EmitHandlerDelegateType` could take a `bool fullyQualifyTypeNames` bool and optionally emit the `global::Namespace` part. But I think this is fine for now.